### ISSUE TITLE
Bug: [no-extra-parens] False positive when used with `satisfies` keyword

### ISSUE THREAD

**Issue by xsjcTony** [typescript-eslint/typescript-eslint#7017]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=C4TwDgpgBAYg9nKBeKByARgQwE6qgHzQDMFUBuAWACgBjOAOwGdgoiAbTAc2SgAoBtVCTioANGiy4AulEaZgAS0ZEFERrAT8pASgB0C+jTYBXACZreQ0trJA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDWsX7RETVBkiJo0DtEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type Foo = 'bar' | 'foo';
const flag = (['foo', 'bar'] satisfies Foo[]).includes('foo');
//           ^                              ^
// Unnecessary parentheses around expression. 2:14 - 2:15
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-extra-parens": "error"
  },
};
```

### Expected Result

There should be no error

### Actual Result

False positive

**Comment by bradzacher** [comment]
We really need to update this rule to use our precedence function instead of using the really disgusting forked logic...!

**Comment by xsjcTony** [comment]
Seems the false positive has gone, but I didn't see any PR merged for it🤣 weird.

**Comment by bradzacher** [comment]
It should be fixed in the latest version of eslint because I upstreamed a fix for it - https://github.com/eslint/eslint/pull/17302

But the playground hasn't been updated to the latest eslint version yet.

**Comment by bradzacher** [comment]
Hi there 👋!

As you may have heard [ESLint core is deprecating its formatting rules](https://eslint.org/blog/2023/10/deprecating-formatting-rules/). We are likewise moving in the same direction and deprecating our formatting rules.

Deprecated rules will be removed in the next major version as their new home is the [eslint-stylistic](https://github.com/eslint-stylistic/eslint-stylistic) project.

If this issue is important to you please consider filling an issue over at eslint-stylistic which will be the new home for stylistic/formatting rules.

Thanks!
